torch/csrc/tensor/python_tensor.cpp

```diff
@@ -0,0 +1,261 @@
+#include "python_tensor.h"
+
+#include <structmember.h>
+#include <mutex>
+#include <pybind11/pybind11.h>
+#include <sstream>
+
+#include "torch/csrc/assertions.h"
+#include "torch/csrc/Exceptions.h"
+#include "torch/csrc/autograd/variable.h"
+#include "torch/csrc/autograd/python_variable.h"
+#include "torch/csrc/autograd/generated/VariableType.h"
+#include "torch/csrc/utils/python_strings.h"
+#include "torch/csrc/utils/tensor_new.h"
+#include "torch/csrc/utils/tensor_types.h"
+
+namespace torch { namespace tensor {
+
+using namespace at;
+using namespace torch::autograd;
+
+struct PyTensorType {
+  PyTypeObject py_type;
+  at::Type* aten_type;
+  bool is_cuda;
+  bool is_sparse;
+  bool is_default;
```

Comment:
comment why you have this.

Reply:
also, a name with "all" is probably better than "default".  is_all_types? is_any_type?